src/mgmt/srv.c

```diff
@@ -410,15 +411,61 @@ process_join_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 }
 
 static int
-create_ownership_props(daos_prop_t **out_prop, char *owner, char *owner_grp)
+ace_strs_to_acl(char **ace_strs, size_t ace_nr, struct daos_acl **acl)
 {
-	char	       *out_owner = NULL;
-	char	       *out_owner_grp = NULL;
-	daos_prop_t    *new_prop = NULL;
+	struct daos_ace	*tmp_aces[ace_nr];
+	struct daos_acl	*tmp_acl = NULL;
+	size_t		i;
+	int		rc;
+
+	if (ace_strs == NULL || ace_nr == 0)
+		return 0; /* nothing to do */
+
+	memset(tmp_aces, 0, sizeof(tmp_aces));
+
+	for (i = 0; i < ace_nr; i++) {
+		rc = daos_ace_from_str(ace_strs[i], &(tmp_aces[i]));
+		if (rc != 0) {
+			D_ERROR("Failed to convert string '%s' to ACE, "
+				"err=%d\n", ace_strs[i], rc);
+			D_GOTO(out, rc);
+		}
+	}
+
+	tmp_acl = daos_acl_create(tmp_aces, ace_nr);
+	if (tmp_acl == NULL) {
+		D_ERROR("Failed to allocate ACL\n");
+		D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	*acl = tmp_acl;
+	rc = 0;
+
+out:
+	for (i = 0; i < ace_nr; i++)
+		daos_ace_free(tmp_aces[i]);
```

Comment:
If acr_nr > 2 and a intermediate daos_ace_from_str() call fails then this looks like it'll call daos_acr_free() with uninitialised data for larger values of i.

Reply:
Changing to allocate on the heap with D_ALLOC_ARRAY to avoid VLAs on the stack. This should also initialize all the array pointers to NULL.